agent/src/com/cloud/agent/AgentShell.java

```diff
@@ -173,6 +194,19 @@ public void setPersistentProperty(String prefix, String name, String value) {
             _storage.persist(name, value);
     }
 
+    @Override
+    public boolean setPropertyAndPersist(String property, String value) {
```

Comment:
Don't use it, there is another way to do that --> call setPersistentProperty, see the method is just above your implementation.